Source/Tools/FEXRootFSFetcher/Main.cpp

```diff
@@ -402,27 +431,96 @@ namespace WebFileFetcher {
     return Exec::ExecAndWaitForResponse(ExecveArgs[0], const_cast<char* const*>(ExecveArgs.data())) == 0;
   }
 
+  struct JsonAllocator {
+    jsonPool_t PoolObject;
+    std::unique_ptr<std::list<json_t>> json_objects;
+  };
+  static_assert(offsetof(JsonAllocator, PoolObject) == 0, "This needs to be at offset zero");
+
+  json_t* PoolInit(jsonPool_t* Pool) {
+    JsonAllocator* alloc = reinterpret_cast<JsonAllocator*>(Pool);
+    alloc->json_objects = std::make_unique<std::list<json_t>>();
+    return &*alloc->json_objects->emplace(alloc->json_objects->end());
+  }
+
+  json_t* PoolAlloc(jsonPool_t* Pool) {
+    JsonAllocator* alloc = reinterpret_cast<JsonAllocator*>(Pool);
+    return &*alloc->json_objects->emplace(alloc->json_objects->end());
+  }
+
   std::vector<FileTargets> GetRootFSLinks() {
     // Decode the filetargets
     std::string Data = DownloadToString(DownloadURL);
-    std::stringstream ss(Data);
+
+    JsonAllocator Pool {
+      .PoolObject = {
+        .init = PoolInit,
+        .alloc = PoolAlloc,
+      },
+    };
+
+    json_t const *json = json_createWithPool(&Data.at(0), &Pool.PoolObject);
+    if (!json) {
+      fprintf(stderr, "Couldn't create json");
+      return {};
+    }
+
+    json_t const* RootList = json_getProperty(json, "v1");
+
+    if (!RootList) {
+      fprintf(stderr, "Couldn't get root list");
+      return {};
+    }
+
     std::vector<FileTargets> Targets;
 
-    while (!ss.eof()) {
+    for (json_t const* RootItem = json_getChild(RootList);
+      RootItem != nullptr;
+      RootItem = json_getSibling(RootItem)) {
+
       FileTargets Target;
-      if (!std::getline(ss, Target.DistroMatch))
-        break;
-      if (!std::getline(ss, Target.VersionMatch))
-        break;
-      if (!std::getline(ss, Target.DistroName))
-        break;
-      if (!std::getline(ss, Target.URL))
-        break;
-      if (!std::getline(ss, Target.Hash))
-        break;
-
-      Targets.emplace_back(Target);
+      Target.DistroName = json_getName(RootItem);
+
+      for (json_t const* DataItem = json_getChild(RootItem);
+        DataItem != nullptr;
+        DataItem = json_getSibling(DataItem)) {
+        auto DataName = json_getName(DataItem);
+
+        if (strcmp(DataName, "DistroMatch") == 0) {
```

Comment:
Minor cleanup, not critical but figured it's worth mentioning since the strcmps add a bunch of noise:

```suggestion
        auto DataName = std::string_view { json_getName(DataItem)};

        if (DataName == "DistroMatch") {
```

Reply:
Resolved.